### PR TITLE
Add documentation for HA cluster support features

### PR DESCRIPTION
**Pull request by barthisrael**
This PR adds documentation for all the features that have been as part
of the work of making it easier to integrate Barman with HA clusters.

We perform the following changes to the web docs / `man`:

* Add `cluster` configuration option to `man 5`
* Add `model` configuration option to `man 5`
* Include the `Model` scope to all options that configuration models support
* Add information about configuration models under the `Integration with cluster
  management systems` section of the web docs
* Add information about configuration models and examples under the `Configuration`
  section of the web docs
* Add documentation about the `barman config-switch` new CLI command in the
  web docs

Some changes to the documentation related with `wal_conninfo` and
`wal_streaming_conninfo` have already been applied through #868, and are only
pending to be merged.

Besides that we applied some standardization on all configuration options in `man`
(`doc/barman.5.d/50-*` files):

* Have an empty new line at the end -- some of them were missing this;
* Have at most 80 characters per line -- some of them were not following
  this rule;
* Have a `Scope: ` line near the end, after the description of the
  configuration option: it specifies which is the scope of the config
  option, i.e., if it can be applied to one or more scopes among
  `Global`, `Server` and `Model`. This information used to be present
  in the `man 5` files already, however they were part of the descritpion.
  Now we have some emphasis with a separate paragraph which follows the
  same standard over all files.

Also, there are more docs to come, once we finish other work related with HA cluster
features which is still pending.

References: BAR-136.

### CODE REVIEW

File: doc/manual/50-feature-details.en.md
Comment (martinmarques) [inline doc/manual/50-feature-details.en.md:902]:
```suggestion
includes the following options:
```

File: doc/manual/50-feature-details.en.md
Comment (martinmarques) [inline doc/manual/50-feature-details.en.md:929]:
```suggestion
That will override the cluster configuration options with the values
```

**Review by martinmarques — CHANGES_REQUESTED** [review]

File: doc/barman.5.d/50-cluster.md
Comment (mikewallace1979) [inline doc/barman.5.d/50-cluster.md:2]:
```suggestion
:   Name of the Barman cluster associated with a Barman server or model. Used
```

File: doc/manual/17-configuration.en.md
Comment (mikewallace1979) [inline doc/manual/17-configuration.en.md:9]: [R5](https://github.com/EnterpriseDB/barman/pull/880/commits/a18f2bda214d20bcb14a0b703968c9d4e512cd29#diff-cb88ce549bfbef5a801899de30b88c00cb5f5544fbad2d1841b20cb24a0f20dcR5) should probably be updated to `There are three types of configuration files in Barman:`

File: doc/manual/17-configuration.en.md
Comment (mikewallace1979) [inline doc/manual/17-configuration.en.md:15]: This was a little hard to parse with the extra comma separated clause - I was thinking it could be reduced to:

```suggestion
Similarly, model configuration files are located in the `/etc/barman.d` directory and must have a `.conf` suffix.
```

I was thinking of adding the `one for each model that can be applied to a server which is part of a cluster` to the end, but I wonder whether one-model-per-file is something we should recommend at all. It might be better to group multiple models for a server (or cluster) in a single file (or maybe not...?).

File: doc/manual/21-preliminary_steps.en.md
Comment (mikewallace1979) [inline doc/manual/21-preliminary_steps.en.md:67]: Is this backtick supposed to be here?

File: doc/manual/17-configuration.en.md
Comment (mikewallace1979) [inline doc/manual/17-configuration.en.md:17]: Could we link to the `barman config-switch` section here?

File: doc/manual/42-server-commands.en.md
Comment (mikewallace1979) [inline doc/manual/42-server-commands.en.md:74]:
```suggestion
server is composed of the configuration of the server plus the overrides applied
```

File: doc/manual/42-server-commands.en.md
Comment (mikewallace1979) [inline doc/manual/42-server-commands.en.md:75]:
```suggestion
by the selected model. Models are particularly useful for clustered environments,
```

File: doc/manual/50-feature-details.en.md
Comment (mikewallace1979) [inline doc/manual/50-feature-details.en.md:891]:
```suggestion
the config models. They can be useful when handling clustered environments, so you
```

File: doc/manual/50-feature-details.en.md
Comment (mikewallace1979) [inline doc/manual/50-feature-details.en.md:918]: It might be worth keeping the conninfo the same apart from the host, just in case a user misinterprets that the `database=postgres` is omitted from conninfo here because it's present in the conninfo in the server configuration.

```suggestion
conninfo = host=pg-node-2 user=barman database=postgres
```

**Review by mikewallace1979 — COMMENTED** [review]
Looks good - I had a few minor wording suggestions but nothing major.

File: doc/manual/17-configuration.en.md
Comment (barthisrael) [reply doc/manual/17-configuration.en.md:15]: I wonder if "one model per file" wouldn't end up bloating the directory 😆 I guess we can just keep the way you suggested above, and add nothing else. What do you think?

File: doc/manual/21-preliminary_steps.en.md
Comment (barthisrael) [reply doc/manual/21-preliminary_steps.en.md:67]: Typo, good catch.

File: doc/manual/17-configuration.en.md
Comment (barthisrael) [reply doc/manual/17-configuration.en.md:17]: I think that makes sense, I'll take a look at adding a ref.

File: doc/manual/50-feature-details.en.md
Comment (barthisrael) [reply doc/manual/50-feature-details.en.md:918]: Makes sense!

File: doc/manual/17-configuration.en.md
Comment (barthisrael) [reply doc/manual/17-configuration.en.md]: Applied.

File: doc/manual/21-preliminary_steps.en.md
Comment (barthisrael) [reply doc/manual/21-preliminary_steps.en.md]: Applied.

File: doc/manual/17-configuration.en.md
Comment (barthisrael) [reply doc/manual/17-configuration.en.md:9]: Applied.

File: doc/manual/17-configuration.en.md
Comment (mikewallace1979) [reply doc/manual/17-configuration.en.md]: Yeah that's fine I think.

**Comment by barthisrael** [comment]
We will merge this once #868 and #873 get merged because this PR includes documentation for those PRs.
We decided to merge this PR and follow-up with documenting future work related with models through the specific PRs that are coming.

**Review by mikewallace1979 — APPROVED** [review]

**Review by gcalacoci — APPROVED** [review]
LGTM

**Comment by barthisrael** [comment]
I'm bypassing protections and merging this anyway (Martin is currently on PTO and his thoughts have been applied to the PR).

**Pull request merged**
